### PR TITLE
[TypeScript][Skill]Main Dialog

### PR DESCRIPTION
**Pull request by matiasroldan6**
### Purpose
Add changes to MainDialog class to port to v1.0

### CODE REVIEW

File: templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts [hunk 1]
```diff
@@ -130,18 +125,18 @@ export class MainDialog extends ComponentDialog {
...
         }
 
         return await super.onContinueDialog(innerDc);
     }
 
     // Runs on every turn of the conversation to check if the conversation should be interrupted.
-    protected async interruptDialog(innerDc: DialogContext): Promise<Boolean> {
-        let interrupted = false;
+    protected async interruptDialog(innerDc: DialogContext): Promise<DialogTurnResult> {
+        let interrupted: DialogTurnResult = {} as DialogTurnResult;
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:139]:
```suggestion
    protected async interruptDialog(innerDc: DialogContext): Promise<DialogTurnResult> | undefined {
        let interrupted: DialogTurnResult | undefined = undefined;
```

Taking into account that [C#](https://github.com/microsoft/botframework-solutions/blob/v1.0/samples/csharp/skill/SkillSample/Dialogs/MainDialog.cs#L118) can return `null` and when it's called to this method always [checks](https://github.com/southworks/botframework-solutions/pull/384/files#diff-031ce6a60277e736381a0f80207d3607R128) if it's undefined or not, I think that we can do this.

What do you think?

File: templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts [hunk 2]
```diff
@@ -155,15 +150,19 @@ export class MainDialog extends ComponentDialog {
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:150]:
```suggestion
```

File: templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts [hunk 3]
```diff
@@ -155,15 +150,19 @@ export class MainDialog extends ComponentDialog {
...
-                        interrupted = true;
+                        if (TurnContextEx.isSkill(innerDc.context)) {
+                            interrupted = await innerDc.endDialog();
+                        }
+                        else {
+                            interrupted = await innerDc.beginDialog(this.initialDialogId);
+                        }
                         break;
                     } 
                     case 'Help': {
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:162]:
```suggestion
```

File: templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts [hunk 4]
```diff
@@ -155,15 +150,19 @@ export class MainDialog extends ComponentDialog {
...
 
                         await innerDc.context.sendActivity(this.templateManager.generateActivityForLocale('HelpCard', innerDc.context.activity.locale));
                         await innerDc.repromptDialog();
-                        interrupted = true;
+                        interrupted = MainDialog.EndOfTurn;
                         break;
                     }
                     case 'Logout': {
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:168]: The following blank space can be deleted, I can't mark this as GitHub rlz, 😅

File: templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts (hunk 1 above)
Comment (matiasroldan6) [reply templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:139]: If this doesn't make a snowball effect and the expected parameter needs to be changed to `DialogTurnResult | undefined` in subsequent methods consuming this, I'm ok

Comment (Batta32) [reply templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:139]: I guess not, as it's correctly validated when it's used. You can double check it!

Comment (matiasroldan6) [reply templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:139]: I'm getting this error when applying this
`TS2322: Type 'DialogTurnResult<any> | undefined' is not assignable to type 'DialogTurnResult<any>'.   Type 'undefined' is not assignable to type 'DialogTurnResult<any>'.`
Perhaps we should find another solution

Comment (Batta32) [reply templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts:139]: Ack, leave it as you did. It's ok!

**Pull request merged**